gemstone/generator/generator.py

```diff
@@ -24,6 +24,21 @@ def add_port(self, name, T):
             raise ValueError(f"{name} is already a port")
         self.ports[name] = PortReference(self, name, T)
 
+    def remove_port(self, port_name: str):
+        # first remove it from self.ports
+        assert port_name in self.ports
+        self.ports.pop(port_name)
+        # then remove any wires connected with it. due to port cloning
+        # the only thing won't change is the port name
+        wires_to_remove = set()
+        for conn1, conn2 in self.wires:
```

Comment:
why not just make the check inside look like
```python
port = self.ports[port_name]
for conn1, conn2 in self.wires:
    if conn1 is port or con2 is port:
        wires_to_remove.add((conn1, conn2))```

Reply:
Because you need to deal with slicing, e.g., `config` is a compound port where you can access the "sub" ports. I don't think the proposed would work. Maybe I'm wrong.